lab_3_ann_retriever/main.py

```diff
@@ -149,6 +191,8 @@ def __init__(self, corpus: list[list[str]]) -> None:
         Args:
             corpus (list[list[str]]): Tokenized documents to vectorize
         """
+        self._corpus = corpus
+        self.build()
```

Comment:
do you really need to call `build` method?

Reply:
![image](https://github.com/user-attachments/assets/906c3cbd-4608-4e22-9be4-1f080ce72b2d)
It won't work without self.build() in __init__. I tried to somehow call build() with the '_vocabulary' but it didn't work :-(

Finally, I completely changed _calculate_tf_idf(self, document: list[str]). There is no self.build() in __init__ now. So, done!